price/api.go

```diff
@@ -1,56 +1,140 @@
+	"github.com/rs/zerolog/log"
 )
 
 type API struct {
-	latestPrices *LatestPrices
+	pricer    *pricing.Pricer
+	jwtSecret string
+	cfger     pricing.ConfigProvider
 }
 
-func NewAPI() *API {
-	// TODO replace with organic pricing logic
-	// current values are copy pasted from node
-	pricePerGiBMax := new(big.Int).SetInt64(500_000_000_000_000_000) // 0.5 MYSTT
-	pricePerHourMax := new(big.Int).SetInt64(180_000_000_000_000)    // 0.0018 MYSTT
-
+func NewAPI(pricer *pricing.Pricer, cfger pricing.ConfigProvider, jwtSecret string) *API {
 	return &API{
-		latestPrices: &LatestPrices{
-			Current: &Prices{
-				PricePerGiB:  new(big.Int).Div(pricePerGiBMax, big.NewInt(2)),
-				PricePerHour: new(big.Int).Div(pricePerHourMax, big.NewInt(2)),
-				ValidUntil:   time.Now().Add(time.Hour).UTC(),
-			},
-			Previous: &Prices{
-				PricePerGiB:  new(big.Int).Div(pricePerGiBMax, big.NewInt(2)),
-				PricePerHour: new(big.Int).Div(pricePerHourMax, big.NewInt(2)),
-				ValidUntil:   time.Now().Add(time.Minute * 30 * -1).UTC(),
-			},
-		},
+		pricer:    pricer,
+		cfger:     cfger,
+		jwtSecret: jwtSecret,
 	}
 }
 
+// LatestPrices returns latest prices
+// @Summary Latest Prices
+// @Description Latest Prices
+// @Product json
+// @Success 200 {array} pricing.LatestPrices
+// @Router /prices [get]
+// @Tags prices
 func (a *API) LatestPrices(c *gin.Context) {
-	c.JSON(200, a.latestPrices)
+	c.JSON(200, a.pricer.GetPrices())
 }
 
-// LatestPrices holds two sets of prices. The Previous should be used in case
-// a race condition between obtaining prices by Consumer and Provider
-// upon agreement
-type LatestPrices struct {
-	Current  *Prices `json:"current"`
-	Previous *Prices `json:"previous"`
+// GetConfig returns the base pricing config
+// @Summary Price config
+// @Description price config
+// @Product json
+// @Success 200 {array} pricing.Config
+// @Router /prices/config [get]
+// @Tags prices
+func (a *API) GetConfig(c *gin.Context) {
+	cfg, err := a.cfger.Get()
+	if err != nil {
+		log.Err(err).Msg("Failed to get config")
+		c.JSON(http.StatusBadRequest, gorest.NewErrResponse(err.Error()))
+		return
+	}
+	c.JSON(http.StatusOK, cfg)
 }
 
-type Prices struct {
-	PricePerHour *big.Int  `json:"price_per_hour"`
-	PricePerGiB  *big.Int  `json:"price_per_gib"`
-	ValidUntil   time.Time `json:"valid_until"`
+// UpdateConfig updates the pricing config
+// @Summary update price config
+// @Description update price config
+// @Product json
+// @Success 202
+// @Param config body pricing.Config true "config object"
+// @Router /prices/config [post]
+// @Tags prices
+func (a *API) UpdateConfig(c *gin.Context) {
+	var cfg pricing.Config
+	if err := c.BindJSON(&cfg); err != nil {
+		c.JSON(http.StatusBadRequest, gorest.NewErrResponse(err.Error()))
+		return
+	}
+
+	err := a.cfger.Update(cfg)
+	if err != nil {
+		log.Err(err).Msg("Failed to update config")
+		c.JSON(http.StatusBadRequest, gorest.NewErrResponse(err.Error()))
```

Comment:
This would also be closer to internal server error than bad request.

Reply:
I'll leave it as a bad request as it returns the validation errors here and I don't feel like complicating the handling here further